systems/plants/@RigidBodyManipulator/forwardKin.m

```diff
@@ -177,18 +195,85 @@
         Jrow_ind = reshape([reshape(1:3*m,3,m);bsxfun(@times,3*m+(1:4)',ones(1,m))],[],1);
         J = Jtmp(Jrow_ind,:);
         
-    end
-    if (nargout>2)
-      if (rotation_type>0)
-        warning('Second derivatives of rotations are not implemented yet.');
       end
-      if isempty(kinsol.ddTdqdq{body_ind})
-        error('you must call doKinematics with the second derivative option enabled');
+      if (nargout>2)
+        if (rotation_type>0)
+          warning('Second derivatives of rotations are not implemented yet.');
+        end
+        if isempty(kinsol.ddTdqdq{body_ind})
+          error('you must call doKinematics with the second derivative option enabled');
+        end
+        ind = repmat(1:3*nq,nq,1)+repmat((0:3*nq:3*nq*(nq-1))',1,3*nq);
+        dJ = reshape(kinsol.ddTdqdq{body_ind}(ind,:)*Tframe*pts,nq^2,[])';
       end
-      ind = repmat(1:3*nq,nq,1)+repmat((0:3*nq:3*nq*(nq-1))',1,3*nq);
-      dJ = reshape(kinsol.ddTdqdq{body_ind}(ind,:)*Tframe*pts,nq^2,[])';
     end
   end
 end
 
-end
\ No newline at end of file
+end
+
+function [x, J, dJ] = forwardKinNew(obj, kinsol, body_or_frame_ind, points, rotation_type, base_ind)
```

Comment:
As I was working on this, I came up with the idea of passing in a boolean argument specifying whether the output Jacobian should act on v or qd, instead of using a separate function forwardKinV. I think that way would be cleaner and would result in a more efficient implementation. I already did this in C++.